### ISSUE TITLE
Output is not configured correctly after reload

### ISSUE THREAD

**Issue by SilverBut** [swaywm/sway#1298]
Reproduction:
1. Write those configurements
```
output HDMI-A-1 resolution 1920x1080 position 0,0
output VGA-1 resolution 1920x1080 position 1920,0
workspace 1 output HDMI-A-1
workspace 2 output VGA-1
```

2. Start sway
3. Reload

Excepted:
a. After step 2, workspace 1 is at the left of workspace 2
b. After step 3, workspace 1 is **still** at the left of workspace 2

Problem:
After step 3, workspace 1 is not at the left of workspace 2. The `position` argument seems ignored.

Version info:
sway version 0.14.0

**Comment by StephenBrown2** [comment]
Why are you setting both positions to 0,0? I believe VGA-1 should be 1920,0 if you wish it to be to the right of HDMI-A-1.

**Comment by SilverBut** [comment]
@StephenBrown2 Sorry for typo. It is 1920,0 in my config file.

**Comment by SilverBut** [comment]
To describe it more clearly: Contents are still at HDMI-A-1 or VGA-1. They are not moved to another display. Seems only the `position` argument is changed.

**Comment by SirCmpwn** [comment]
Provide a log, please.

**Comment by SilverBut** [comment]
@SirCmpwn @StephenBrown2 

Sorry for such a late response. Log is [here]:https://gist.github.com/SilverBut/ee367166a713d8ad736e658b48d0de14

Now version is 0.14.0 and problem still exists.

**Comment by SirCmpwn** [comment]
Odd. What happens when you position eDP-1 at 3840?

**Comment by SirCmpwn** [comment]
(I know it's disabled)

**Comment by SilverBut** [comment]
Seems like some variables are not calculated properly. 

Before reload: 
```
08/13/17 20:53:34 - [layout.c:975] Arranging output 'VGA-1' at 1920.000000,0.000000
08/13/17 20:53:34 - [debug_log.c:98] Arranging layout for 0x77c34ab520 (OUTPUT 'VGA-1')
08/13/17 20:53:34 - [layout.c:968] Arranging layout for 0x77c34ab520 VGA-1 1920.000000x1080.000000+1920.000000,0.000000
08/13/17 20:53:34 - [debug_log.c:98] Arranging layout for 0x77c34ab6a0 (WORKSPACE '2 VGA')
08/13/17 20:53:34 - [layout.c:968] Arranging layout for 0x77c34ab6a0 2 VGA 1920.000000x1080.000000+1920.000000,0.000000
08/13/17 20:53:34 - [layout.c:1008] -> Found panel for this workspace: 0x0, position: 0
08/13/17 20:53:34 - [layout.c:1030] Arranging workspace '2 VGA' at 0.000000, 0.000000
08/13/17 20:53:34 - [layout.c:1148] Arranging 0x77c34ab6a0 horizontally
08/13/17 20:53:34 - [layout.c:975] Arranging output 'HDMI-A-1' at 0.000000,0.000000
```

After reload: 
```
08/13/17 20:53:36 - [layout.c:975] Arranging output 'VGA-1' at 3840.000000,0.000000
08/13/17 20:53:36 - [debug_log.c:98] Arranging layout for 0x77c34ab520 (OUTPUT 'VGA-1')
08/13/17 20:53:36 - [layout.c:968] Arranging layout for 0x77c34ab520 VGA-1 1920.000000x1080.000000+3840.000000,0.000000
08/13/17 20:53:36 - [debug_log.c:98] Arranging layout for 0x77c34ab6a0 (WORKSPACE '2 VGA')
08/13/17 20:53:36 - [layout.c:968] Arranging layout for 0x77c34ab6a0 2 VGA 1920.000000x1057.000000+0.000000,23.000000
08/13/17 20:53:36 - [layout.c:1008] -> Found panel for this workspace: 1920x23, position: 0
08/13/17 20:53:36 - [layout.c:1030] Arranging workspace '2 VGA' at 0.000000, 23.000000
08/13/17 20:53:36 - [layout.c:1148] Arranging 0x77c34ab6a0 horizontally
08/13/17 20:53:36 - [layout.c:975] Arranging output 'HDMI-A-1' at 5760.000000,0.000000
```

**Comment by SirCmpwn** [comment]
Yeah, it looks like it's reassigning them new automatic positions, maybe?

**Comment by SilverBut** [comment]
> Odd. What happens when you position eDP-1 at 3840?

@SirCmpwn  I'd rather not try it. eDP-1 is completely broken now, physically.

**Comment by SirCmpwn** [comment]
Not like this will make it worse. Give it a shot.

**Comment by SilverBut** [comment]
Here is my opinion: both screen is 1920x1080, and the new arrangement have two problems:

1. `assert( VGA-1 == HDMI-A-1 + 1920 )`  failed
2. `assert( min(VGA-1.x, HDMI-A-1.x) == 0 )` failed

Should be treated separetly. 

> Not like this will make it worse. Give it a shot.

Will try it later...but let me read the related code first.

**Comment by SilverBut** [comment]
@SirCmpwn One OT question: many logs are from `debug_log.c:98`. Is this work as designed?

**Comment by SirCmpwn** [comment]
Yeah that's fine.

**Comment by hillyu** [comment]
Similar problem happened to me.
In my config:
```
output eDP-1 res 1366x768 pos 1920,180
output HDMI-A-2 res 1920x1080 pos 0,0
```
before _reload_ output is configured as defined in the config file.

after reload the position is automatically moved as if the original ones are occupying the spots and newly created ones are shifted to the right.
```
~ » swaymsg -t get_outputs                                                                           
Output eDP-1
  Geometry: 1366x768 @ 3286,180
  Scale factor: 1x
  Workspace: 10 
Output HDMI-A-2
  Geometry: 1920x1080 @ 4652,0
  Scale factor: 1x
  Workspace: 1 
```

**Comment by SirCmpwn** [comment]
I'm closing issues that are resolved by the yet-unreleased sway 1.0.
